data/data/aws/master/main.tf

```diff
@@ -84,7 +84,7 @@ resource "aws_instance" "master" {
   user_data            = "${var.user_data_ign}"
 
   vpc_security_group_ids      = ["${var.master_sg_ids}"]
-  associate_public_ip_address = "${var.public_endpoints}"
+  associate_public_ip_address = "true"
```

Comment:
This can be the boolean `true`, can't it?  Docs for [the setting][1] and [boolean values][2].  They have some caveats for variable inputs, but here we're setting a parameter, so I think `true` (without the wrapping quotes) makes the most sense.

[1]: https://www.terraform.io/docs/providers/aws/r/instance.html#associate_public_ip_address
[2]: https://www.terraform.io/docs/configuration/variables.html#booleans